policy-controller/k8s/index/src/pod.rs

```
@@ -72,48 +72,44 @@ pub(crate) fn port_names(spec: &Option<k8s::PodSpec>) -> HashMap<String, PortSet
-                                        .and_then(NonZeroU16::try_from)
-                                    {
-                                        let paths = http_probes.entry(port).or_default();
-                                        paths.insert(path.clone());
-                                    }
-                                    break;
-                                }
-                            }
-                        }
-                    }
+pub(crate) fn pod_http_probes(pod: &k8s::PodSpec) -> PortMap<HashSet<String>> {
+    let mut probes = PortMap::<HashSet<String>>::default();
+    for (port, path) in pod.containers.iter().flat_map(container_http_probe_paths) {
+        probes.entry(port).or_default().insert(path);
+    }
+    probes
+}
+
+fn container_http_probe_paths(container: &k8s::Container) -> Vec<(NonZeroU16, String)> {
+    fn find_by_name(name: &str, ports: &[k8s::ContainerPort]) -> Option<NonZeroU16> {
+        for port in ports {
+            if let Some(ref n) = port.name {
+                if n.eq_ignore_ascii_case(name) {
+                    let p = u16::try_from(port.container_port).ok()?;
+                    return p.try_into().ok();
                 }
             }
         }
+        None
+    }
+
+    fn get_port(port: &k8s::IntOrString, container: &k8s::Container) -> Option<NonZeroU16> {
+        match port {
+            k8s::IntOrString::Int(p) => u16::try_from(*p).ok()?.try_into().ok(),
+            k8s::IntOrString::String(n) => find_by_name(n, &*container.ports.as_ref()?),
+        }
     }
-    http_probes
+
+    (container.liveness_probe.iter())
+        .chain(container.readiness_probe.iter())
+        .chain(container.startup_probe.iter())
+        .filter_map(|p| {
+            let probe = p.http_get.as_ref()?;
+            let port = get_port(&probe.port, container)?;
+            let path = probe.path.clone().unwrap_or_else(|| "/".to_string());
```

As mentioned in #8961, I don't think we should panic if a path is unset.